File: frontend/src/services/authService.js
import { api } from './api.js';

export const authService = {
  async register(payload) {
    const { data } = await api.post('/auth/register', payload);
    return data;
  },

  async login(payload) {
    const { data } = await api.post('/auth/login', payload);
    return data;
  },

  async me() {
    const { data } = await api.get('/users/me');
    return data;
  },

  async updateProfile(payload) {
    const { data } = await api.put('/users/me', payload);
    return data;
  },

  async changePassword(payload) {
    await api.put('/users/password', payload);
  },
};

File: backend/src/main/java/com/cartcraft/service/AuthService.java
package com.cartcraft.service;

import com.cartcraft.dto.auth.AuthResponse;
import com.cartcraft.dto.auth.LoginRequest;
import com.cartcraft.dto.auth.RegisterRequest;
import com.cartcraft.entity.User;
import com.cartcraft.exception.ConflictException;
import com.cartcraft.repository.UserRepository;
import com.cartcraft.security.JwtService;
import lombok.RequiredArgsConstructor;
import org.springframework.security.authentication.AuthenticationManager;
import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class AuthService {
    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtService jwtService;
    private final AuthenticationManager authenticationManager;

    @Transactional
    public AuthResponse register(RegisterRequest request) {
        if (userRepository.existsByEmail(request.email())) {
            throw new ConflictException("Email is already registered");
        }

        User user = User.builder()
                .name(request.name().trim())
                .email(request.email().trim().toLowerCase())
                .password(passwordEncoder.encode(request.password()))
                .role(request.role())
                .build();

        User saved = userRepository.save(user);
        return toAuthResponse(saved);
    }

    public AuthResponse login(LoginRequest request) {
        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
                request.email().trim().toLowerCase(),
                request.password()
        ));
        User user = userRepository.findByEmail(request.email().trim().toLowerCase())
                .orElseThrow();
        return toAuthResponse(user);
    }

    private AuthResponse toAuthResponse(User user) {
        return new AuthResponse(
                jwtService.generateToken(user),
                user.getId(),
                user.getName(),
                user.getEmail(),
                user.getRole()
        );
    }
}
